Eatery/Controllers/Eateries/EateriesViewController.swift

```diff
@@ -500,184 +531,179 @@ extension EateriesViewController: UICollectionViewDataSource {
-    
-    func searchBarCancelButtonClicked(_ searchBar: UISearchBar) {
-        searchBar.text = ""
-        processEateries()
-        collectionView.reloadData()
-        searchBar.resignFirstResponder()
-        searchBar.setShowsCancelButton(false, animated: true)
-    }
-    
+
     func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
-        if let searchText = searchBar.text, !searchText.isEmpty {
-            processEateries()
-            collectionView.reloadData()
-        }
-        searchBar.setShowsCancelButton(false, animated: true)
         searchBar.resignFirstResponder()
     }
-    
+
     func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
-        processEateries()
-        collectionView.reloadData()
+        reloadEateries()
     }
-    
-    
-    // MARK: - CLLocationManagerDelegate Methods
-    
-    func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
-        userLocation = locations.last
 
-        for cell in collectionView.visibleCells.compactMap({ $0 as? EateryCollectionViewCell }) {
-            cell.userLocation = userLocation
-        }
-    }
-    
-    func locationManager(_ manager: CLLocationManager, didFailWithError error: Error) {
-        print("Location Manager Error: \(error)")
+    func searchBarShouldEndEditing(_ searchBar: UISearchBar) -> Bool {
+        return true
     }
-    
+
 }
 
+// MARK: - Filter Bar
+
 extension EateriesViewController: FilterBarDelegate {
-    func updateFilters(filters: Set<Filter>) {
-        self.filters = filters
-        processEateries()
-        collectionView.reloadData()
+
+    func filterBar(_ filterBar: FilterBar, selectedFiltersDidChange newValue: [Filter]) {
+        reloadEateries()
     }
+
 }
 
-extension EateriesViewController: UIViewControllerPreviewingDelegate {
+// MARK: - Failed To Load View
 
-    func previewingContext(_ previewingContext: UIViewControllerPreviewing, viewControllerForLocation location: CGPoint) -> UIViewController? {
-        let collectionViewPoint = view.convert(location, to: collectionView)
-        
-        guard let indexPath = collectionView.indexPathForItem(at: collectionViewPoint),
-            let cell = collectionView.cellForItem(at: indexPath) as? EateryCollectionViewCell else {
-                return nil
-        }
-        
-        let menuViewController = MenuViewController(eatery: eatery(for: indexPath), delegate: self, userLocation: userLocation)
-        menuViewController.preferredContentSize = CGSize(width: 0.0, height: 0.0)
-        cell.transform = .identity
-        previewingContext.sourceRect = collectionView.convert(cell.frame, to: view)
-        return menuViewController
+extension EateriesViewController: EateriesFailedToLoadViewDelegate {
+
+    func eateriesFailedToLoadViewPressedRetryButton(_ eftlv: EateriesFailedToLoadView) {
+        delegate?.eateriesViewControllerDidPressRetryButton(self)
     }
+
+}
+
```

Comment:
I feel like we should also implement detecting touchesBegan and dismiss the keyboard when users are searching. Pressing the search button after maybe even clicking the search bar accidentally shouldn't be the only way to exit searching.

Reply:
The user can also scroll to get out of searching, but this is an issue for another PR.